test/sys/test_sockopt.rs

```diff
@@ -236,6 +236,33 @@ fn test_so_tcp_keepalive() {
     }
 }
 
+#[test]
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[cfg_attr(qemu, ignore)]
+fn test_get_mtu() {
+    use nix::sys::socket::{bind, connect, SockaddrIn};
+    use std::net::SocketAddrV4;
+    use std::str::FromStr;
+
+    let std_sa = SocketAddrV4::from_str("127.0.0.1:4001").unwrap();
```

Comment:
Could you simplify this test by using `socketpair` instead of `socket`, `bind`, `connect`?

Reply:
This fails with EOPNOTSUPP:
> The specified protocol does not permit creation of socket pairs.

Should I revert it to socket-bind-connect?

Reply:
Either that, or switch to SockType::Stream .

Reply:
turning protocol into none and domain into unix seems to give a valid socketpair, only IP_MTU is not supported on it

I'll turn it back to socket-bind-connect.